push_test/pushclient.py

```diff
@@ -0,0 +1,335 @@
+import asyncio
+import base64
+import json
+import uuid
+import sys
+import traceback
+
+import aiohttp
+import websockets
+
+
+class PushException(Exception):
+    pass
+
+
+def log_msg(out=sys.stdout, **msg):
+    print(json.dumps(msg), file=out)
+
+
+class PushClient(object):
+    """Smoke Test the Autopush push server"""
+    def __init__(self, args, loop, tasks=[]):
+        self.config = args
+        self.loop = loop
+        self.connection = None
+        self.pushEndpoint = None
+        self.channelID = None
+        self.notifications = []
+        self.uaid = None
+        self.recv = []
+        self.tasks = tasks
+        self.output = None
+
+    def _next_task(self):
+        try:
+            task = self.tasks.pop(0)
+            return "cmd_" + task[0], task[1]
+        except IndexError:
+            return "cmd_done", {}
+
+    async def run(self,
+                  server="wss://push.services.mozilla.com",
+                  tasks=[]):
+        """Connect to a remote server and execute the tasks
+
+        :param server: URL to the Push Server
+
+        """
+        if tasks:
+            self.tasks = tasks
+        if not self.connection:
+            await self.cmd_connect(server)
+        while len(self.tasks):
+            cmd, args = self._next_task()
+            try:
+                await getattr(self, cmd)(**args)
```

Comment:
There's no catch here of the AttributeError that will occur if the command is spelled wrong. It might be less error-prone if the commands were all constants, and there was a dict in the class that mapped the constants to the class functions for each command.